auraed/src/cells/cell_service/validation.rs

```diff
@@ -294,3 +294,224 @@ impl From<ValidatedExecutable> for super::executables::ExecutableSpec {
         Self { name, command: c, description }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_cell_type_empty_cpu_valid() {
+        let validated =
+            CellValidator::validate_cpu(None, "field", Some("parent"));
+        assert!(validated.is_ok());
+        assert!(validated.unwrap().is_none());
+    }
+
+    #[test]
+    fn test_cell_type_cpu_valid() {
+        let validated = CellValidator::validate_cpu(
+            Some(CpuController { weight: Some(1000), max: None }),
+            "field",
+            Some("parent"),
+        );
+        assert!(validated.is_ok());
+        let inner = validated.unwrap();
+        assert!(inner.is_some());
+        let controller = inner.unwrap();
+        assert_eq!(controller.weight, Some(Weight::new(1000)));
+        assert_eq!(controller.max, None);
+    }
+
+    #[test]
+    fn test_cell_type_cpu_weight_too_small() {
+        let validated = CellValidator::validate_cpu(
+            Some(CpuController { weight: Some(0), max: None }),
+            "field",
+            Some("parent"),
+        );
+        assert!(validated.is_err());
+    }
+
+    #[test]
+    fn test_cell_type_cpu_weight_too_large() {
+        let validated = CellValidator::validate_cpu(
+            Some(CpuController { weight: Some(20000), max: None }),
```

Comment:
Is it better to test the first invalid value (i.e., 10001)?

Reply:
a QA tester walks into a bar, orders 1 beer, 10001 beers, 20000 beers, NaN beers...